shared/src/domain-logic/opportunityFields.ts

```diff
@@ -10,9 +10,33 @@ export const OPPORTUNITY_FIELDS: (keyof ProfileData)[] = [
 
 export const REPORT_FIELDS: (keyof ProfileData)[] = ["dateOfFormation"];
 
+export const FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION: (keyof ProfileData)[] = ["municipality"];
+export const FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION: (keyof ProfileData)[] = ["homeBasedBusiness"];
+export const ALL_FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION = [
+  ...FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION,
+  ...REPORT_FIELDS,
+];
+export const ALL_FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION = [
+  ...FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION,
+  ...REPORT_FIELDS,
+];
 export const FIELDS_FOR_PROFILE = [...REPORT_FIELDS, ...OPPORTUNITY_FIELDS];
-export const getFieldsForProfile = (legalStructureId?: string): (keyof ProfileData)[] => {
-  return FIELDS_FOR_PROFILE.filter((field) => {
+
+export const getFieldsForProfile = (profileData: ProfileData): (keyof ProfileData)[] => {
+  if (profileData.businessPersona === "FOREIGN" && profileData.foreignBusinessType === "NEXUS") {
+    return profileData.nexusLocationInNewJersey
+      ? filterByLegalStructure(ALL_FIELDS_FOR_FOREIGN_NEXUS_NJ_LOCATION, profileData.legalStructureId)
+      : filterByLegalStructure(ALL_FIELDS_FOR_FOREIGN_NEXUS_NOT_NJ_LOCATION, profileData.legalStructureId);
+  } else {
+    return filterByLegalStructure(FIELDS_FOR_PROFILE, profileData.legalStructureId);
+  }
+};
+
+export const filterByLegalStructure = (
```

Comment:
[sand] I think most of these functions & variables are only used in this file, right? For anything that is intended only for use in this file, which I think would include this function and most (all?) of the constant variables definitions, they should not include the `export` keyword